### PR TITLE
fix "Closing split bill doesn't refocus on composer" on web/desktop

### PR DESCRIPTION
**Pull request by aimane-chnaif**
### Details
In [PopoverMenu](https://github.com/Expensify/App/blob/main/src/components/PopoverMenu/index.js) component, `props.onItemSelected(item);` should be called first. And then `item.onSelected();` should be called after onModalHide callback.
This is already done in native side: https://github.com/Expensify/App/blob/main/src/components/PopoverMenu/index.native.js
We no longer need platform specific handling for this component.
So remove current [index.js](https://github.com/Expensify/App/blob/main/src/components/PopoverMenu/index.js) and rename [index.native.js](https://github.com/Expensify/App/blob/main/src/components/PopoverMenu/index.native.js) to index.js

### Fixed Issues
$ https://github.com/Expensify/App/issues/10888
PROPOSAL: https://github.com/Expensify/App/issues/10888#issuecomment-1245652236

### Tests
1. Login with any account on desktop or web.
2. Click on any report in LHN.
3. Click on + icon on left end of composer.
4. Click one of Request money / Split money / Split bill.
5. Click cross icon to close this modal page.
6. Verify that composer input is focused automatically

NOTE: It's intended behavior to prevent focus on touchable devices (native, mWeb) after navigating back from modal screens. reference: https://github.com/Expensify/App/blob/fbf0334713e7e8d73507e30a8b12c941d9b46670/src/pages/home/report/ReportActionCompose.js#L168-L174

- [x] Verify that no errors appear in the JS console

### QA Steps
1. Login with any account on desktop or web.
2. Click on any report in LHN.
3. Click on + icon on left end of composer.
4. Click one of Request money / Split money / Split bill.
5. Click cross icon to close this modal page.
6. Verify that composer input is focused automatically

- [ ] Verify that no errors appear in the JS console

#### PR Author Checklist
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

<details>
<summary><h4>PR Reviewer Checklist</h4>

The reviewer will copy/paste it into a new comment and complete it after the author checklist is completed
</summary>

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [ ] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

</details>

#### Web

https://user-images.githubusercontent.com/96077027/196853192-5006f8d1-5069-4f0e-b474-a514dd622ce7.mov

#### Mobile Web - Chrome

https://user-images.githubusercontent.com/96077027/196853143-7bbc51e5-bf59-42fd-8025-a08ca8b82550.mp4

#### Mobile Web - Safari

https://user-images.githubusercontent.com/96077027/196853167-006e6561-9eec-45c8-9715-c1cd18b0094f.mov

#### Desktop

https://user-images.githubusercontent.com/96077027/196853089-29ff0d9e-517c-4f4f-a5e9-662d2181fe43.mov

#### iOS

https://user-images.githubusercontent.com/96077027/196853128-8577adf7-6db1-4428-88de-8bfa23769918.mov

#### Android

https://user-images.githubusercontent.com/96077027/196853038-84c1cb6e-c040-42cb-8784-e28f94ac4ddc.mp4

### CODE REVIEW

**Comment by mananjadhav** [comment]
@Luke9389 @aimane-chnaif I was just about to test and while checking the code locally, I think we can further refactor this.

We have `BasePopoverMenu` currently, only because we have two files, `index.js` and `index.native.js`. With us merging all code into `index.js`, `BasePopoverMenu` is redundant. I feel we should merge `BasePopoverMenu` itno `PopoverMenu/index.js` 

What do you folks think?

**Comment by mananjadhav** [comment]
@aimane-chnaif Can you implement the changes requested?


cc - @Luke9389

**Comment by mananjadhav** [comment]
thanks @aimane-chnaif. Will need 1-2 days to test this as it'll required a good amount of regression suite to test.

**Comment by mananjadhav** [comment]
Thanks for the changes and patience on this one @aimane-chnaif. Changes look good and test well. I agree with your comment that for mobile devices (app and mweb) we don't want to refocus as it'll popup the keyboard.

@Luke9389 All yours

🎀 👀 🎀 
C+ reviewed

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files
    - [x] I verified any copy / text that was added to the app is correct English and approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots


#### Web

https://user-images.githubusercontent.com/3069065/199849463-1c1540bd-3919-445c-a8e8-ad1582f9d217.mov

#### Mobile Web - Chrome

https://user-images.githubusercontent.com/3069065/199849521-0aceb1a3-11a8-43cc-afb5-d930c9370b81.mov

#### Mobile Web - Safari

https://user-images.githubusercontent.com/3069065/199849584-facda024-79c9-4a0f-8475-4ddef80e722e.mov

#### Desktop

https://user-images.githubusercontent.com/3069065/199849610-f0659a32-d9ed-4b56-b737-9c2e92925950.mov

#### iOS

https://user-images.githubusercontent.com/3069065/199849669-94561efc-74dc-4022-97a8-be87f2976b43.mov

#### Android

https://user-images.githubusercontent.com/3069065/199849736-9eb42a64-c879-4c23-87db-398ca3ef0179.mov

**Review by Luke9389 — COMMENTED** [review]
I'll be coming around to this one tomorrow. Thanks for the changes.

**Comment by mananjadhav** [comment]
@Luke9389 Quick bump on this one

**Comment by mananjadhav** [comment]
@Luke9389 did you get a chance to look at this PR?

**Comment by Luke9389** [comment]
Oh man, missed it. Thanks for the bump(s)

File: src/components/PopoverMenu/index.js [hunk 1]
```diff
@@ -1,30 +1,147 @@
...
+
+    /**
+     * @param {Number} index
+     */
+    updateFocusedIndex(index) {
+        this.setState({focusedIndex: index});
+    }
+
+    resetFocusAndHideModal() {
+        this.updateFocusedIndex(-1); // Reset the focusedIndex on modal hide
```
[the remark this answers is not in the document]
Comment (Luke9389) [reply src/components/PopoverMenu/index.js:92]: Is this function used anywhere?

**Review by Luke9389 — CHANGES_REQUESTED** [review]

File: src/components/PopoverMenu/index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Luke9389) [reply src/components/PopoverMenu/index.js:92]: oh duh, thanks

File: src/components/PopoverMenu/index.js [hunk 2]
```diff
@@ -1,30 +1,147 @@
...
+
+    /**
+     * Set the item's `onSelected` action to fire after the menu popup closes
+     * @param {{onSelected: function}} item
+     */
+    selectItem(item) {
+        this.onMenuHide = () => {
+            item.onSelected();
+
+            // Clean up: open and immediately cancel should not re-trigger the last action
```
[the remark this answers is not in the document]
Comment (Luke9389) [reply src/components/PopoverMenu/index.js:59]: Yeah I like that change. Go ahead and commit it and test it out!

**Comment by Luke9389** [comment]
@aimane-chnaif did you test this on all platforms (multiple pop-overs, not just the bill split one)?

**Comment by mananjadhav** [comment]
@Luke9389 @aimane-chnaif Is there anything pending here? Then I can retest with the latest changes and update the checklist again.

**Review by mananjadhav — APPROVED** [review]

**Comment by mananjadhav** [comment]
@Luke9389  Did you get a chance to look at the previous comments?

**Comment by mananjadhav** [comment]
@Luke9389 This PR is blocked by your review and approval.

**Review by Luke9389 — APPROVED** [review]

**Comment by marcaaron** [comment]
Hi this PR has been linked to a regression here: https://github.com/Expensify/App/issues/13435

I think since this is a lower level change more care could have been applied to test all the existing popovers. In fact, that is what this checkbox is asking us to do.

>I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like Avatar, I verified the components using Avatar have been tested & I retested again)

@mananjadhav @aimane-chnaif please keep this in mind in the future. If we are making changes to something that is used multiple places it's a good idea to quickly check all the usages of that thing, thanks!

**Pull request merged**